docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
 
  * Submit your calculation to the AiiDA daemon:
 
-   .. code-block:: bash
+    .. code-block:: bash
 
-       verdi run submit.py
+        verdi run submit.py
 
 After this, use ``verdi process list`` to monitor the status of the calculations.
 
 See :ref:`topics:processes:usage:launching` and :ref:`topics:processes:usage:monitoring` for more details.
 
-.. todo::
 
-    .. _how-to:run-codes:caching:
 
-    title: Using caching to save computational resources
+.. _how-to:run-codes:caching:
+
+How to save computational resources using caching
+=================================================
+
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Maybe you run a great number of complex workflows in high-throughput that each may repeat the same calculation, or you may have to restart an entire workflow that failed somewhere half-way through.
+Since AiiDA stores the full provenance of each calculation, it can detect whether a calculation has been run before and, instead of running it again, simply reuse its outputs, thereby saving valuable computational resources.
+This is what we mean by **caching** in AiiDA.
```

Comment:
I think there should be a fat warning (disclaimer) with a link to the limitations here, because – unless I misunderstood – there are some cases in which AiiDA can have a false cache hit, which would pose a significant risk to use caching. Since caching is not 100% safe, users need to know about this right from the start. I assume that's also the reason why it's not enabled by default?

Reply:
depends on what you mean by "false" cache hit. If it is really due to two calculations _not_ being equal, but AiiDA erroneously thinking them to be equal, that is bad. However, I think these cases are unlikely, although I cannot exclude them of course. The bigger question though is what I wrote in point 2) of my response to your next comment.

Reply:
A false cache hit in this context would be the code using a cached result when in fact the result would be different on recalculation. This is what I consider the worst case scenario and what I think we have to warn the user about.